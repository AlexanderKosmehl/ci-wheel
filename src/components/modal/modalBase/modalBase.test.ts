import {
  describe, expect, it, vi,
} from 'vitest';
import generateModal from './modalBase';

describe('generateModal', () => {
  const content = document.createElement<'div'>('div');
  content.textContent = 'Content';

  const onClose = vi.fn(() => {});

  const modal = generateModal({
    title: 'Title',
    content,
    onClose,
  });

  const closeButton = modal.querySelector<HTMLButtonElement>('.closeButton');

  it('generates component correctly', () => {
    expect(modal).toMatchSnapshot();
  });

  it('adds working onClose callback', () => {
    if (!closeButton) throw Error('No closeButton rendered!');

    closeButton.click();
    expect(onClose).toBeCalled();
  });
});
