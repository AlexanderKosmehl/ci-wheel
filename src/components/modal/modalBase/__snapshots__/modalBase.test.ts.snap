// Vitest Snapshot v1

exports[`generateModal > generates component correctly 1`] = `
<div
  class="container"
>
  <div
    class="content"
  >
    <div
      class="header"
    >
      <h2
        class="headerText"
      >
        Title
      </h2>
      <button
        class="button closeButton"
      >
        <img
          class="icon"
          src="/src/assets/close-icon.svg"
        />
      </button>
    </div>
    <div>
      Content
    </div>
  </div>
</div>
`;
